pkg/apis/kubeone/validation/validation.go

```diff
@@ -461,7 +462,13 @@ func ValidateAddons(o *kubeone.Addons, fldPath *field.Path) field.ErrorList {
 		return allErrs
 	}
 	if o.Enable && len(o.Path) == 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), "", ".addons.path must be specified"))
+		// Check if only embedded addons are being used; path is not required for embedded addons
+		areEmbeddedAddons, err := addons.AreEmbeddedAddons(o.Addons)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, "", "failed to read embedded addons directory"))
+		} else if !areEmbeddedAddons {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("path"), "", ".addons.path must be specified when using non-embedded addon(s)"))
+		}
```

Comment:
What if someone provides something like this:

```yaml
addons:
  enable: true
```

Should we reject that as invalid or allow it? As far as I see, the current implementation will allow it, but does such configuration make sense at all?

Reply:
TBH if the user specifies that is he doing something wrong? I don't think so but yes the configuration doesn't make sense either. It's something somewhat benign though.
WDYT @kron4eg? Let's weigh-in and make a quick decision.

Reply:
I discussed this with @kron4eg and we concluded that the following config should be treated as invalid:

```yaml
...
addons:
  enable: true
```

In other words, if the user:

* doesn't enable any embedded addon
* doesn't provide a path (in which case no custom addon can be deployed)

we should reject such config, i.e. validation should fail. That's because such addons config is not useful or doing anything at all.

@ahmedwaleedmalik Please add a validation case for this.

Reply:
Okay :+1:

Reply:
Resolved, PTAL.